### PR TITLE
codegen and publish java client lib

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [x] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,27 @@
 jdk:
-  - oraclejdk8
```
Comment (dvoet) [inline .travis.yml:2]: sorry for the reformatting of this file, travis did it... the env variables and any line with `client` in it is new

Comment (helgridly) [reply .travis.yml:2]: https://github.com/broadinstitute/sam/pull/280/files?utf8=%E2%9C%93&diff=split&w=1 this link will make the whitespace differences in the GH view less painful

File: scripts/gen_java_client.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+set -e
+
+docker run --rm -v ${PWD}:/local openapitools/openapi-generator-cli generate -i /local/src/main/resources/swagger/api-docs.yaml -g java -o /local/codegen_java --api-package org.broadinstitute.dsde.workbench.client.sam.api --model-package org.broadinstitute.dsde.workbench.client.sam.model
+
+cd codegen_java
+
+sbt test
```
Comment (helgridly) [inline scripts/gen_java_client.sh:7]: I'm confused -- why are you re-testing here?

Comment (dvoet) [reply scripts/gen_java_client.sh:7]: the code generator makes tests too... thought I'd run them

File: .travis.yml [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+  - secure: uFfbRLBp0iaHAIPQr1UomDWPWD6pgAnd5xSOlGlnJdxQQEaR96jgIl9DuBTmwowLDuldQxIn5C+OUDU69TtobqfR4bf90K3431aF3DSS92x1DI5QxtpKJbutNS7vObz7OIRyvbRfGzu0XXsI2ceEv12GHH1u3pyL1z7+GePxI8fIYacSAQrCTs63LJaLurVpUkBN55K02o//dz1uf/cyvsfxwWossGM2T/RHyAiIimXP/43tdAUeRAz5VRpi/YbBQxjUAoGo1hDF+ZmlyK0KVHSI4kCK48GlypmxiYDar/7g7xwmZM4IF5UO4+iKySfzD4kIZCyHe4pok0L4RltunnbRdsEyPkdKPKisprFLD28htbi+lJcumEz4WU0t5FEGK9WnHcO9D1/P6xPUJgUPfoWqRb3McpFe0uTmRKEwq/jGdvImZKf6krO3jlDeCulSKAbLx1zRI0+OlvVXYVmMtyig4H0SF/H6garlV1SReH1xJ7ta8v8QtnFzc4ZstD7KoUHsRpM7girQZ0Qxg/scTDvxV38e8dLpe1V0dkgdBE8kzlXSmzwDLqXOXDrYRVYmjLKO3Kb1G5ZKjK3jNcDsYAhTlj18fcsCQpYeCeU3npHt4lvPXPTi3kzx4rcJSHnHCR9sJK7mtqlCUWvldyCQ2yBVmRZLseTaxvJjuEL1kSM=
```
Comment (ansingh7115) [inline .travis.yml:27]: What are these again?

Comment (dvoet) [reply .travis.yml:27]: artifactory user and password

**Comment by dvoet** [comment]
jenkins retest - looks like a docker issue

**Comment by dvoet** [comment]
jenkins retest - looks like a docker issue

File: scripts/gen_java_client.sh (hunk 2 above)
Comment (dvoet) [reply scripts/gen_java_client.sh:7]: the codegen_java dir has its own sbt files so this is running those tests, not the top level tests

**Comment by dvoet** [comment]
jenkins retest

**Review by ansingh7115 — APPROVED** [review]
Looks good - thank you for updating this. @MatthewBemis Leo should also be updated to use the new client

**Comment by dvoet** [comment]
jenkins retest - docker pull issue

**Pull request merged**